packages/docusaurus-theme-classic/src/theme/TagsListByLetter/index.tsx

```diff
@@ -6,11 +6,12 @@
  */
 
 import React from 'react';
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
 import {listTagsByLetters, type TagLetterEntry} from '@docusaurus/theme-common';
 import Tag from '@theme/Tag';
 import type {Props} from '@theme/TagsListByLetter';
-
 import styles from './styles.module.css';
+import type {CharMap} from '@docusaurus/types';
```

Comment:
I wanna put the `CharMap` import before the `styles`, but my Webstorm just won't ....

Reply:
Oh, why? (And yes—I assume our ESLint rule wants you to put the type import before CSS import)

Reply:
all the eslint import ruler are copied here, maybe need be checked again:

```js
    'import/extensions': OFF,
    // This rule doesn't yet support resolving .js imports when the actual file
    // is .ts. Plus it's not all that useful when our code is fully TS-covered.
    'import/no-unresolved': [
      OFF,
      {
        // Ignore certain webpack aliases because they can't be resolved
        ignore: [
          '^@theme',
          '^@docusaurus',
          '^@generated',
          '^@site',
          '^@testing-utils',
        ],
      },
    ],
    'import/order': [
      WARNING,
      {
        groups: [
          'builtin',
          'external',
          'internal',
          ['parent', 'sibling', 'index'],
          'type',
        ],
        pathGroups: [
          {pattern: '@jest/globals', group: 'builtin', position: 'before'},
          {pattern: 'react', group: 'builtin', position: 'before'},
          {pattern: 'fs-extra', group: 'builtin'},
          {pattern: 'lodash', group: 'external', position: 'before'},
          {pattern: 'clsx', group: 'external', position: 'before'},
          // 'Bit weird to not use the `import/internal-regex` option, but this
          // way, we can make `import type { Props } from "@theme/*"` appear
          // before `import styles from "styles.module.css"`, which is what we
          // always did. This should be removable once we stop using ambient
          // module declarations for theme aliases.
          {pattern: '@theme/**', group: 'internal'},
          {pattern: '@site/**', group: 'internal'},
          {pattern: '@theme-init/**', group: 'internal'},
          {pattern: '@theme-original/**', group: 'internal'},
        ],
        pathGroupsExcludedImportTypes: [],
      },
    ],
    'import/prefer-default-export': OFF,
```

Reply:
To be clear—I'm not sure _which_ way I enforced (although I wrote that config myself so I know I enforced one side). You just need to make sure ESLint is happy. Looking at the lint run, seems it's fine currently. I was just asking if your editor is reporting an ESLint warning or something else.

Reply:
Yes, I would check by myself why would this happen: `ESLint: `@docusaurus/types` import should occur after import of `./styles.module.css`(import/order)`.

Reply:
It's quite suspicious that the `@docusaurus/**` is left ... Now the warning has gone away.